### ISSUE TITLE
Better handling of comments when adding to an array

### ISSUE THREAD

**Issue by oguyotot** [mikefarah/yq#547]
Description of the current behavior (yq 3.4.0):

```
$ cat sample.yaml 
yq:
  # a comment
  - rocks: true

$ yq r sample.yaml 'yq[0]'
# a comment
rocks: true
```

=> so far so good. Notice that yq is reading the array entry's comment, which is great. But then:

```
$ yq r sample.yaml 'yq[0]' | yq w -f - sample.yaml 'yq[+]'
yq:
  # a comment
  - rocks: true
  - # a comment
    rocks: true
```

=> when adding the entry with the comment to the array, the format is not the one expected. Instead, I would have expected this:

```
yq:
  # a comment
  - rocks: true
  # a comment
  -  rocks: true
```
Thanks!

**Comment by mikefarah** [comment]
the comments are handled by the underlying go-yaml parser - and yeah it's not great :/

Feel free to raise an issue on that project, however they have commented that they make a 'best effort' approach and not aiming to perfectly handle comments and whitespaces

**Comment by mikefarah** [comment]
This has been fixed in `yq` version 4.3.1
